File: quizzes.js
const admin = require('firebase-admin');
const fs = require('fs-extra');
const yaml = require('yamljs');

admin.initializeApp({
    credential: admin.credential.cert(require('./credentials.json')),
});
const db = admin.firestore();

const quizzes = [
    'angular-basics',
    'flutter-scroll',
    'cf-basics',
    'cf-triggers',
    'firebase-perf',
    'firestore-basics',
    'flutter-basics',
    'flutter-containers',
    'flutter-flex',
    'flutter-gestures',
    'flutter-material',
    'flutter-scroll',
    'js-basics',
    'js-variables',
    'rxjs-basics',
    'ts-basics'
]

const update = async(quizId) => {

    const json = yaml.load(`quizzes/${quizId}.yaml`);

    const ref = db.collection('quizzes').doc(quizId);

    await ref
        .set(json, { merge: true })
        .catch(e => console.log(`'${quizId}' upload failed with error: ${e.details}`));

    console.log(`'${quizId}' is DONE`);
}

for (const quiz of quizzes) {
    update(quiz);
}

File: quizzes.py
import firebase_admin
from firebase_admin import credentials, firestore
import yaml
import asyncio

# Initialize Firebase Admin SDK
cred = credentials.Certificate('./credentials.json')
firebase_admin.initialize_app(cred)
db = firestore.client()

quizzes = [
    'angular-basics',
    'flutter-scroll',
    'cf-basics',
    'cf-triggers',
    'firebase-perf',
    'firestore-basics',
    'flutter-basics',
    'flutter-containers',
    'flutter-flex',
    'flutter-gestures',
    'flutter-material',
    'flutter-scroll',
    'js-basics',
    'js-variables',
    'rxjs-basics',
    'ts-basics'
]

async def update(quiz_id):
    with open(f'quizzes/{quiz_id}.yaml', 'r') as file:
        yaml_data = yaml.safe_load(file)

    print(yaml_data)

    ref = db.collection('quizzes').document(quiz_id)

    await ref.set(yaml_data, merge=True)

    print('DONE')

# Loop through quizzes and update Firestore
async def main():
    tasks = [update(quiz) for quiz in quizzes]
    await asyncio.gather(*tasks)

if __name__ == "__main__":
    asyncio.run(main())
